src/common/components/vanilla/button/button.css

```diff
@@ -107,15 +107,23 @@ $neutral-border-color: $dark-grey;
   box-sizing: content-box;
 }
 
+.hasSpinner {
+  position: relative;
+}
+
+.hasSpinner .text {
+  /* hide text without chaning button width */
+  opacity: 0;
+}
+
 .spinner {
-  display: block;
-  margin: 0;
-  padding: 1px 0 0 0;
-  margin-right: 0.5em;
-  height: 21px;
-  width: 21px;
-  float: left;
-  box-sizing: content-box;
+  position: absolute;
+  height: 20px;
+  width: 20px;
+  left: 50%;
+  top: 50%;
```

Comment:
Thanks for suggestion, I tend to forget we already have all major browsers supporting stuff like this.